core/src/main/web/app/helpers/helper.js

```diff
@@ -305,6 +305,108 @@
+          var bkRenderer = new marked.Renderer();
+          bkRenderer.link = function(href, title, text) {
+            var prot;
+            if (this.options.sanitize) {
+              try {
+                prot = decodeURIComponent(unescape(href))
+                    .replace(/[^\w:]/g, '')
+                    .toLowerCase();
+              } catch (e) {
+                return '';
+              }
+              //jshint ignore:start
+              if (prot.indexOf('javascript:') === 0 || prot.indexOf('vbscript:') === 0) {
+                //jshint ignore:end
+                return '';
+              }
+            }
+            var out = '<a href="' + href + '"';
+            if (title) {
+              out += ' title="' + title + '"';
+            }
+            out += ' target="_blank"'; // < ADDED THIS LINE ONLY
+            out += '>' + text + '</a>';
+            return out;
+          };
+
+          bkRenderer.paragraph = function(text) {
+            // Allow users to write \$ to escape $
+            return marked.Renderer.prototype.paragraph.call(this, text.replace(/\\\$/g, '$'));
+          };
+          this.bkRenderer = bkRenderer;
+        }
+
+        var doktex = function(markdownFragment) {
+          try {
+            renderMathInElement(markdownFragment[0], {
+              delimiters: [
+                {left: "$$", right: "$$", display: true},
+                {left: "$", right:  "$", display: false},
+                {left: "\\[", right: "\\]", display: true},
+                {left: "\\(", right: "\\)", display: false}
+              ]
+            });
+          } catch(err) {
+            bkHelper.show1ButtonModal(err.message + '<br>See: ' +
+                '<a target="_blank" href="http://khan.github.io/KaTeX/">KaTeX website</a> and its ' +
+                '<a target="_blank" href="https://github.com/Khan/KaTeX/wiki/Function-Support-in-KaTeX">' +
+                'list of supported functions</a>.',
+                "KaTex error");
+          }
+        };
+
+        var markIt = function(content) {
+          var markdownFragment = $('<div>' + content + '</div>');
+          doktex(markdownFragment);
+          var escapedHtmlContent = angular.copy(markdownFragment.html());
+          markdownFragment.remove();
+          var unescapedGtCharacter = escapedHtmlContent.replace(/&gt;/g, '>');
+          var result = marked(unescapedGtCharacter, {
+            gfm: true,
+            renderer: bkRenderer
+          });
+          markupDeferred.resolve(result);
+        };
+
+        var results = [], re = /{{([^}]+)}}/g, text;
+
+        while (text = re.exec(cellContent)) {
+          if (results.indexOf(text) === -1)
+            results.push(text);
+        }
+
+        var evaluateCode = function (index) {
+
+          if (index === results.length) {
+            markIt(cellContent);
+          } else {
+            bkHelper.evaluateCode("JavaScript", results[index][1]).then(
```

Comment:
I think this will currently fail on Publications. We don't initialize Beaker app so `getCurrentApp` defined as (and needed to eval code):
```js
    var getCurrentApp = function() {
      return bkCoreManager.getBkApp();
    };
```
will be undefined. If we could somehow explicitly evaluate this code with plain javascript `eval()` without going through Beaker's evaluators it would work...